src/java/src/test/android/app/src/androidTest/java/ai/onnxruntime_genai/example/javavalidator/SimpleTest.kt

```diff
@@ -1,42 +1,121 @@
 package ai.onnxruntime.genai.example.javavalidator
 
 import ai.onnxruntime.genai.*
-
-import android.os.Build;
+import android.os.Build
 import android.util.Log
 import androidx.test.ext.junit.rules.ActivityScenarioRule
 import androidx.test.ext.junit.runners.AndroidJUnit4
 import androidx.test.platform.app.InstrumentationRegistry
-import com.microsoft.appcenter.espresso.Factory
-import com.microsoft.appcenter.espresso.ReportHelper
 import org.junit.*
 import org.junit.runner.RunWith
+import java.io.File
+import java.io.FileOutputStream
 import java.io.IOException
 import java.util.*
 
+
 private const val TAG = "ORTGenAIAndroidTest"
 
 @RunWith(AndroidJUnit4::class)
 class SimpleTest {
     @get:Rule
     val activityTestRule = ActivityScenarioRule(MainActivity::class.java)
 
-    @get:Rule
-    var reportHelper: ReportHelper = Factory.getReportHelper()
-
     @Before
-    fun Start() {
-        reportHelper.label("Starting App")
-        Log.println(Log.INFO, TAG, "SystemABI=" + Build.SUPPORTED_ABIS[0])
+    fun start() {
+        Log.i(TAG, "SystemABI=" + Build.SUPPORTED_ABIS[0])
     }
 
-    @After
-    fun TearDown() {
-        reportHelper.label("Stopping App")
+    @Throws(IOException::class)
+    private fun copyModelFromAssets(): String {
+        // NOTE: We have to read from the app's assets (app/src/main/assets) and write the the app's filesDir.
+        // The unit test's context cannot be used. you'll get mysterious errors like File.mkdirs() will return false and
+        // assertManager.open() throws even if the filename is valid if you try and use it.
+        // Test context is InstrumentationRegistry.getInstrumentation().targetContext.
```

Comment:
Important to note this restriction as it will apply to any app using GenAI. 

Any model included in the app will need to be copied to the app's `filesDir` so that it's not part of a compressed archive and the individual files (e.g. the config) are accessible from the native code.